forloop_modules/globals/local_variable_handler.py

```diff
@@ -57,12 +59,35 @@ def __init__(self):
         self.variables_to_be_created_in_subpipeline = []
         self.variables={}
         self.variable_uid_variable_dict={} #ANALOGY of dicts in GLC, new implementation - contains nodes in API -> reflecting status of server nodes via API
-        
+
+        self._handler_mode: Literal["initial_variable", "variable"] = "initial_variable"
```

Comment:
On first look, I didnt understand what this is -> Either add comment which explains - "Allows to switch between manipulating with intial_variables and variables"
Or rename it to e.g. "initial_variables_toggle" (bool), "initial_variables_or_variables_mode" (literal), "variable_handler_mode" + consider adding the comment as well

Word "handler" is too ambiguous inside of our codebase (I would think of pipeline function handlers)